grid-demo/src/app/angular/lazy-loading-rows/lazy-loading-rows.component.html

```diff
@@ -1 +1,25 @@
-Angular - Lazy Loading Rows
+<clr-datagrid
+  *ngIf="{ data: data | async }; let data"
+  [clrDgLoading]="!data.data"
+  [clrDgSelected]="[]"
+  [clrDgRowSelection]="true"
+  style="height: 500px"
+>
+  <clr-dg-column *ngFor="let column of columns" [clrDgField]="column.field">
+    <ng-container *clrDgHideableColumn>{{ column.displayName }}</ng-container>
+  </clr-dg-column>
+
+  <ng-container *ngFor="let vm of data.data?.vms">
+    <clr-dg-row *ngIf="vm" [clrDgItem]="vm">
```

Comment:
How can we get here at all if `vm` is Falsy?
We expect some data array that contains null/undefined items?

Reply:
Yes, that's how I've implemented the smart-seek lazy-load features. The grid actually contains all the items, but most are null until they are scrolled into view.